src/context/products/Context.jsx

```diff
@@ -0,0 +1,71 @@
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+  useReducer,
+} from "react";
+import { productsReducer } from "./Reducer";
+import axios from "axios";
+import {
+  getResultantProducts,
+  sortProducts,
+  filterProductsByCategory,
+  filterProductsWithMinPrice,
+  filterByRating,
+  filterOutOutOfStockProducts,
+  showFastDeliveryProducts,
+} from "./utils";
+
+const initialState = {
+  sortBy: "",
+  categories: [],
+  minPrice: 0,
+  rating: 0,
+  showOnlyInStock: false,
+  fastDelivery: false,
+};
+
+const ProductsContext = createContext();
+
+const useProducts = () => useContext(ProductsContext);
+
+const ProductsProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(productsReducer, initialState);
+
+  const [products, setProducts] = useState([]);
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const { data } = await axios.get("/api/products");
+        setProducts(data?.products);
+      } catch (e) {
+        console.log("error", e);
```

Comment:
Maybe replace console log with an alert message.

Reply:
this is temporary for now will refactor later